Firestore/Swift/Source/PropertyWrapper/FirestoreQueryObservable.swift

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import FirebaseFirestore
+
+@available(iOS 14.0, macOS 11.0, macCatalyst 14.0, tvOS 14.0, watchOS 7.0, *)
+internal class FirestoreQueryObservable<T>: ObservableObject {
+  @Published var items: T
+
+  private let firestore = Firestore.firestore()
+  private var listener: ListenerRegistration?
+
+  private var setupListener: (() -> Void)!
+
+  internal var preventUpdates = false
+  internal var configuration: FirestoreQuery<T>.Configuration {
+    didSet {
+      if !preventUpdates {
+        removeListener()
+        setupListener()
+      }
```

Comment:
WDYT about changing `preventUpdate` to `shouldUpdate`/`shouldUpdateListener` (and flipping all assigned values to account for renaming).

There is something about `!preventUpdates` that reads like a double negative and confuses me.
Proposing:
```swift
// --snip --
  internal var shouldUpdateListener = true
  internal var configuration: FirestoreQuery<T>.Configuration {
    didSet {
      if shouldUpdateListener {
        removeListener()
        setupListener()
      }
// --snip --
```
And, later in `projectError(_ error:)`:
```swift
  private func projectError(_ error: Error?) {
    shouldUpdateListener = false
    configuration.error = error
    shouldUpdateListener = true
  }
```

Reply:
The reason why I chose `preventUpdates` is very specific - we need to actively _prevent_ any updates to `configuration.error` to trigger a SwiftUI state update  cycle (as this results in an endless loop, no matter the state of the error).

You will notice that updates will only be blocked when assigning `error`.

That's why I chose `prevent`. I also considered `blockUpdates`, `stopBubblingUp` (just briefly), and others. If anyone has any better names, LMK.

Reply:
Aha I see the endless loop issue now, that's smart 👍  `preventUpdates` LGTM

Reply:
After a bit of deliberation, changed the name to `shouldUpdateListener`, used a `guard` , and added a comment to make this clearer.

Reply:
Latest revisions look great! Thanks 👍